examples/ASCollectionView/Sample/ViewController.m

```diff
@@ -15,50 +15,48 @@
 #import "SupplementaryNode.h"
 #import "ItemNode.h"
 
+@interface ViewController ()
+@property (nonatomic, strong) ASCollectionView *collectionView;
+@property (nonatomic, strong) NSArray *data;
+@end
+
 @interface ViewController () <ASCollectionViewDataSource, ASCollectionViewDelegateFlowLayout>
-{
-  ASCollectionView *_collectionView;
-  NSArray *_data;
-}
 
 @end
 
 
 @implementation ViewController
 
-#pragma mark -
-#pragma mark UIViewController.
+- (void)dealloc
+{
+  self.collectionView.asyncDataSource = nil;
```

Comment:
is this actually necessary since these are weak references?  It may actually be, because of the proxy, but not sure.